app/models/phrase_block.rb

```diff
@@ -0,0 +1,54 @@
+# frozen_string_literal: true
+
+# == Schema Information
+#
+# Table name: phrase_blocks
+#
+#  id          :bigint(8)        not null, primary key
+#  phrase      :text             not null
+#  filter_type :integer          default("text"), not null
+#  whole_word  :boolean          default(TRUE), not null
+#  created_at  :datetime         not null
+#  updated_at  :datetime         not null
+#
+class PhraseBlock < ApplicationRecord
+  enum filter_type: { text: 0, regexp: 1 }, _suffix: :type
+
+  validates :phrase, presence: true
+  validates :phrase, regex: true, if: :regexp_type?
+
+  after_commit :invalidate_cache!
+
+  def to_log_human_identifier
+    if regexp_type?
+      "/#{phrase}/"
```

Comment:
Should this include the flag, given that's how it's executed?
```suggestion
      "/#{phrase}/i"
```